PhysicsTools/PatAlgos/plugins/PATPackedCandidateProducer.cc

```diff
@@ -129,7 +135,7 @@ void pat::PATPackedCandidateProducer::produce(edm::StreamID, edm::Event& iEvent,
     iEvent.getByToken( PuppiCandsMap_, puppiCandsMap );
     edm::Handle<std::vector< reco::PFCandidate > > puppiCands;
     iEvent.getByToken( PuppiCands_, puppiCands );
-    std::vector<int> mappingPuppi(puppiCands->size());
+    std::vector<int> mappingPuppi(usePuppi_ ? puppiCands->size() : 0);
```

Comment:
can we put all the getByToken of the puppi's and the loop of line 147-149 in an `if (usePuppi_)`? (at that point, the isValid will no longer be needed)
also, I would replace the other `isValid()` calls to `usePuppi_` (e.g. line 266)